### PR TITLE
log unhandled errors during request to console

### PR DESCRIPTION
**Pull request by phryneas**
This logs an error message to the console in the case an unhandled error is thrown during a request.

This should happen in user-provided code, either in
* `transformResponse`
* `baseQuery`
* `fetchBaseQuery`: in the `prepareHeaders` or `validateStatus` option.

As is will lead to `providesTags`/`invalidatesTags`not being taken into account for that request, we probably should warn this to the console with a very red warning.

### CODE REVIEW

File: packages/toolkit/src/query/core/buildThunks.ts [hunk 1]
```diff
@@ -313,6 +313,15 @@ export function buildThunks<
       if (error instanceof HandledError) {
         return rejectWithValue(error.value, { baseQueryMeta: error.meta })
       }
+      if (
+        typeof process !== 'undefined' &&
+        process.env.NODE_ENV === 'development'
+      ) {
+        console.log(`An unhandled error occured processing a request for the endpoint ${arg.endpointName}.
```
Comment (msutkowski) [inline packages/toolkit/src/query/core/buildThunks.ts:320]:
```suggestion
        console.warn(`An unhandled error occured processing a request for the endpoint ${arg.endpointName}.
```

File: packages/toolkit/src/query/core/buildThunks.ts [hunk 2]
```diff
@@ -313,6 +313,15 @@ export function buildThunks<
...
+In the case of an unhandled error, no tags will be "provided" or "invalidated".`, error)
+      } else {
+        console.log(error)
```
Comment (msutkowski) [inline packages/toolkit/src/query/core/buildThunks.ts:323]:
```suggestion
        console.error(error)
```

**Review by msutkowski — CHANGES_REQUESTED** [review]
I think we should use either `console.warn` or `console.error` here. I personally prefer using `warn` when providing a message like this though.

File: packages/toolkit/src/query/core/buildThunks.ts (hunk 1 above)
Comment (phryneas) [reply packages/toolkit/src/query/core/buildThunks.ts:320]: Actually I think both of these should be `console.error`. Dunno what I was (not) thinking there.

Comment (msutkowski) [reply packages/toolkit/src/query/core/buildThunks.ts:320]: You put the intention in the description though! 🏆

**Comment by phryneas** [comment]
Since the message is just an addition to the error itself, I've gone for the error.

**Comment by msutkowski** [comment]
Can you get a test to run for this by any chance? I gave it a shot, and couldn't get this to log. I tried `throw`ing in all mentioned places.

**Comment by phryneas** [comment]
Apparently I'm too stupid for a `console.log` PR... I'll try in the evening again.

**Comment by phryneas** [comment]
It works in the tests and in the CSB build so I'll assume there was some hiccup on your side and get this merged :)
image

**Pull request merged**